extensions/mssql/src/main.ts

```diff
@@ -83,6 +85,21 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
 	return createMssqlApi(appContext);
 }
 
+async function setClientQueryExecutionOptions() {
+	vscode.workspace.onDidOpenTextDocument(async (textDocument: vscode.TextDocument) => {
```

Comment:
I'm not clear why this setting is being special handled compare to all the other query options?

also, if you're hooking document open do you also need to special process files that are opened automatically on startup.  I don't think those trigger this event.

Reply:
The only other query option we're currently using is SqlCMD mode and that has its own toggle on the UI. If we can think of a better way to exposing this to the user I can change how we handle it.

Reply:
What about all the ANSI SQL options and other query options?  Those are getting set on the queries we execute.

Reply:
Let me take a look - do you know where we apply that?

Reply:
...in case it's not clear, I'm basically asking why are we handling the default value for this setting in the UI layer rather than providing the desired default value in the tools service, then allow the user to override with a different value in the UI if desired.  Ideally we should have a uniform way of handling all our query execution settings.